kubernetes/config-maps/nginx-proxy/configs/http.conf.j2

```diff
@@ -20,15 +20,19 @@ upstream {{ current_golem_messages_version }} {
      server nginx-proxy.default.svc.cluster.local;
 }
 
+{% if previous_golem_messages_version is not none and previous_golem_messages_version is defined %}
 upstream {{ previous_golem_messages_version }} {
      server {{ concent_versions[previous_concent_version].cluster_address }}:443;
 }
+{% endif %}
 
 # Map to different upstream backends based on the `X-Golem-Messages` http header
 map $http_x_golem_messages $version {
      ""          "http://{{ current_golem_messages_version }}";
      "~^{{ current_golem_messages_version }}.*$"   "http://{{ current_golem_messages_version }}";
+     {% if previous_golem_messages_version is not none and previous_golem_messages_version is defined %}
      "~^{{ previous_golem_messages_version }}.*$"  "https://{{ previous_golem_messages_version }}";
+     {% endif %}
```

Comment:
This part should have a comment explaining why one connection is over HTTPS and the other over HTTP.